From e7aa70f46b6192ebccec40377775fc215418fc3a Mon Sep 17 00:00:00 2001
From: Dhanavandhana Kannan <quic_dhanavan1@quicinc.com>
Date: Thu, 13 Feb 2025 10:54:49 +0530
Subject: [PATCH] QSDK:PROTOCOL:[WAR]:sigma_dut: Set DTIM period to a default
 value of 1

Certain MBSSID grouping certification testcases are failing due to
critical update flag not set to 0 when BPCC is not incremented with
a DTIM period of 2. Critical update flag will remain as 1 until the DTIM
count is 0 with a DTIM period of 2(i.e critical update flag will reset
only after two beacons when DTIM period is 2).

As WFA script expects the critical update flag to reset to 0
immmediately after the next beacon, explicitly defining the DTIM period
to a value of 1 for both mbssid and cohosted bss when MBSSID grouping
is enabled.

Signed-off-by: Dhanavandhana Kannan <quic_dhanavan1@quicinc.com>
---
 ap.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/ap.c
+++ b/ap.c
@@ -11592,6 +11592,7 @@ skip_vht_parameters_set:
 						fprintf(f, "wpa_passphrase=%s\n", dut->mbss_set[j].conf[i].psk);
 					}
 
+					fprintf(f, "dtim_period=1\n");
 					if (dut->program == PROGRAM_EHT || mode == AP_11be)
 						fprintf(f, "mld_ap=1\n");
 
@@ -11818,6 +11819,8 @@ skip_vht_parameters_set:
 					        dut->cohosted[dut->cohost_iface].conf[i].psk);
 				}
 
+				if (dut->multi_mbssid == VALUE_ENABLED)
+					fprintf(f, "dtim_period=1\n");
 				fprintf(f, "mld_ap=1\n");
 			}
 		}
